src/lib/ext-activitylog.js

```diff
@@ -3,26 +3,71 @@ import { save } from './save-load.js';
 class Model {
   constructor() {
     this.logs = [];
+    this.filter = {
+      id: [],
+      viewType: [],
+      type: [],
+    };
   }
 
-  addNewLogs(logs) {
-    this.logs.push(...logs);
+  addNewLog(log) {
+    this.logs.push(log);
+  }
+
+  addFilter({ logKey, valueEquals }) {
+    this.filter[logKey].push(valueEquals);
+  }
+
+  removeFilter({ logKey, valueEquals }) {
+    const logKeyIndex = this.filter[logKey].indexOf(valueEquals);
+    if (logKeyIndex > -1) {
+      this.filter[logKey].splice(logKeyIndex, 1);
+    }
   }
 }
 
 class View {
   constructor() {
-    this.logTable = document.querySelector('log-view');
-    this.saveLogBtn = document.getElementById('saveLogBtn');
+    this.logView = document.querySelector('log-view');
+    this.saveLogBtn = document.querySelector('#saveLogBtn');
     this.notice = document.querySelector('.notice');
 
-    this.saveLogBtn.addEventListener('click', function () {
-      this.dispatchEvent(new CustomEvent('savelog'));
-    });
+    this.extFilter = document.querySelector(
+      'filter-container[filter-key="id"]'
+    );
+    this.viewTypeFilter = document.querySelector(
+      'filter-container[filter-key="viewType"]'
+    );
+    this.apiTypeFilter = document.querySelector(
+      'filter-container[filter-key="type"]'
+    );
+
+    this.saveLogBtn.addEventListener('click', this);
   }
 
-  addTableRows(logs) {
-    this.logTable.addNewRows(logs);
+  handleEvent(event) {
+    if (event.type === 'click') {
+      switch (event.target) {
+        case this.saveLogBtn:
+          this.saveLogBtn.dispatchEvent(new CustomEvent('savelog'));
+          break;
+        default:
+          throw new Error(`wrong event target - ${event.target.tagName}`);
+      }
+    } else {
+      throw new Error(`wrong event type - ${event.type}`);
+    }
+  }
+
+  handleNewLog(object) {
```

Comment:
Looks unused, so remove it. Also, `object` is a very bad parameter name, please try to choose more meaningful variable names if you can.